sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantsLineListing/layout/LineListingLayout.java

```diff
@@ -0,0 +1,360 @@
+
+		sharedInformationComponent.addComponent(sharedInformationBar);
+		addComponent(sharedInformationComponent);
+
+		LineListingSection lineComponent = new LineListingSection(Captions.lineListingNewCasesList);
+
+		eventParticipantLines = new ArrayList<>();
+
+		EventParticipantLineLayout line = buildNewLine(lineComponent);
+		eventParticipantLines.add(line);
+		lineComponent.addComponent(line);
+		lineComponent.setSpacing(false);
+		addComponent(lineComponent);
+
+		UserProvider currentUserProvider = UserProvider.getCurrent();
+
+		if (currentUserProvider != null && currentUserProvider.hasRegionJurisdictionLevel()) {
+			RegionReferenceDto userRegion = currentUserProvider.getUser().getRegion();
+			region.setValue(userRegion);
+			region.setVisible(false);
+			updateDistricts(userRegion);
+		} else {
+			region.setItems(FacadeProvider.getRegionFacade().getAllActiveByServerCountry());
+		}
+
+		HorizontalLayout actionBar = new HorizontalLayout();
+		Button addLine = ButtonHelper.createIconButton(Captions.lineListingAddLine, VaadinIcons.PLUS, e -> {
+			EventParticipantLineLayout newLine = buildNewLine(lineComponent);
+			eventParticipantLines.add(newLine);
+			lineComponent.addComponent(newLine);
+
+			if (eventParticipantLines.size() > 1) {
+				eventParticipantLines.get(0).getDelete().setEnabled(true);
+			}
+		}, ValoTheme.BUTTON_PRIMARY);
+
+		actionBar.addComponent(addLine);
+		actionBar.setComponentAlignment(addLine, Alignment.MIDDLE_LEFT);
+
+		addComponent(actionBar);
+
+		HorizontalLayout buttonsPanel = new HorizontalLayout();
+		buttonsPanel.setMargin(false);
+		buttonsPanel.setSpacing(true);
+		buttonsPanel.setWidth(100, Unit.PERCENTAGE);
+
+		Button cancelButton = ButtonHelper.createButton(Captions.actionDiscard, event -> closeWindow());
+
+		buttonsPanel.addComponent(cancelButton);
+		buttonsPanel.setComponentAlignment(cancelButton, Alignment.BOTTOM_RIGHT);
+		buttonsPanel.setExpandRatio(cancelButton, 1);
+
+		Button saveButton =
+			ButtonHelper.createButton(Captions.actionSave, event -> saveCallback.accept(getEventParticipantLineDtos()), ValoTheme.BUTTON_PRIMARY);
+
+		buttonsPanel.addComponent(saveButton);
+		buttonsPanel.setComponentAlignment(saveButton, Alignment.BOTTOM_RIGHT);
+		buttonsPanel.setExpandRatio(saveButton, 0);
+
+		addComponent(buttonsPanel);
+		setComponentAlignment(buttonsPanel, Alignment.BOTTOM_RIGHT);
+	}
+
+	private LinkedList<LineDto<EventParticipantDto>> getEventParticipantLineDtos() {
+		return eventParticipantLines.stream().map(line -> {
+			EventParticipantLineDto eventParticipantLineDto = line.getBean();
+			LineDto<EventParticipantDto> result = new LineDto<>();
+
+			final EventParticipantDto eventParticipant =
+				EventParticipantDto.build(eventDto.toReference(), UserProvider.getCurrent().getUserReference());
+			eventParticipant.setInvolvementDescription(eventParticipantLineDto.getInvolvementDescription());
+
+			EventDto eventDto = FacadeProvider.getEventFacade().getByUuid(this.eventDto.getUuid());
```

Comment:
`this.eventDto` is the actual event dto so there is no need to load it again and again for each line

Reply:
removed